tests/Unit/NumericalAlgorithms/LinearOperators/Test_CoefficientTransforms.cpp

```diff
@@ -0,0 +1,125 @@
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+#include "Utilities/Algorithm.hpp"
+#include "Utilities/Gsl.hpp"
+
+// These tests generate a function `u_expected` from a linear superposition of
+// the basis functions, which are then transformed to spectral space
+// (u_coeff) and the coefficients compared with the coefficients set in physical
+// space. The functions are then transformed back (u_back) and compared to the
+// original function, `u_expected`.
+namespace {
+template <Spectral::Basis Basis, Spectral::Quadrature Quadrature, size_t Dim>
+void check_transforms(
+    const Mesh<Dim>& mesh,
+    const std::vector<std::array<size_t, Dim>>& coeffs_to_include) noexcept {
+  CAPTURE(Basis);
+  CAPTURE(Quadrature);
+
+  const auto logical_coords = logical_coordinates(mesh);
+  DataVector u_expected(mesh.number_of_grid_points(), 0.0);
+  for (const auto& coeffs : coeffs_to_include) {
+    DataVector basis_function = Spectral::compute_basis_function_value<Basis>(
+        coeffs[0], get<0>(logical_coords));
+    for (size_t dim = 1; dim < Dim; ++dim) {
+      basis_function *= Spectral::compute_basis_function_value<Basis>(
+          gsl::at(coeffs, dim), logical_coords.get(dim));
+    }
+    u_expected += basis_function;
+  }
+
+  const ModalVector u_coeffs = to_modal_coefficients(u_expected, mesh);
+  const DataVector u_back = to_nodal_coefficients(u_coeffs, mesh);
+
+  for (IndexIterator<Dim> index_it(mesh.extents()); index_it; ++index_it) {
+    if (alg::found(coeffs_to_include, index_it->indices())) {
+      CHECK(u_coeffs[index_it.collapsed_index()] == approx(1.0));
+    } else {
+      CHECK(u_coeffs[index_it.collapsed_index()] == approx(0.0));
+    }
+  }
+  CHECK_ITERABLE_APPROX(u_expected, u_back);
+}
+
+template <Spectral::Basis Basis, Spectral::Quadrature Quadrature>
+void test_1d() noexcept {
+  // Start at 1st order so we are independent of the minimum number of
+  // coefficients.
+  for (size_t order = 1; order < Spectral::maximum_number_of_points<Basis>;
+       ++order) {
+    CAPTURE(order);
+    const Mesh<1> mesh(order + 1, Basis, Quadrature);
+    check_transforms<Basis, Quadrature>(mesh, {{{{order}}}});
+    check_transforms<Basis, Quadrature>(mesh, {{{{order}}, {{order / 2}}}});
+    if (order > 4) {
+      check_transforms<Basis, Quadrature>(mesh, {{{{order}}, {{order / 3}}}});
+    }
+  }
+}
+
+template <Spectral::Basis Basis, Spectral::Quadrature Quadrature>
+void test_2d() noexcept {
+  // Start at one higher order so we can drop one order in the y-direction.
+  for (size_t order = Spectral::minimum_number_of_points<Basis, Quadrature> + 1;
+       order < Spectral::maximum_number_of_points<Basis>; ++order) {
+    CAPTURE(order);
+    const Mesh<2> mesh({{order + 1, order}}, Basis, Quadrature);
+    check_transforms<Basis, Quadrature>(mesh, {{{{order, order - 1}}}});
+    check_transforms<Basis, Quadrature>(
+        mesh, {{{{order, order - 1}}, {{order / 2, order / 2}}}});
+    if (order > 4) {
```

Comment:
If my guess in the 1D case was correct, I think you do not need the `if` statement here: the `order/3` term should always be a different coefficient than the `order, order-1` term. (Same comment in 3D)